trio/_core/_tests/test_multierror.py

```diff
@@ -34,46 +36,47 @@ def __eq__(self, other: object) -> bool:
         return self.code == other.code
 
 
-async def raise_nothashable(code):
+async def raise_nothashable(code: int) -> NoReturn:
     raise NotHashableException(code)
 
 
-def raiser1():
+def raiser1() -> NoReturn:
     raiser1_2()
 
 
-def raiser1_2():
+def raiser1_2() -> NoReturn:
     raiser1_3()
 
 
-def raiser1_3():
+def raiser1_3() -> NoReturn:
     raise ValueError("raiser1_string")
 
 
-def raiser2():
+def raiser2() -> NoReturn:
     raiser2_2()
 
 
-def raiser2_2():
+def raiser2_2() -> NoReturn:
     raise KeyError("raiser2_string")
 
 
-def raiser3():
+def raiser3() -> NoReturn:
     raise NameError
 
 
-def get_exc(raiser):
+def get_exc(raiser: Callable[[], NoReturn]) -> BaseException:
     try:
         raiser()
     except Exception as exc:
         return exc
+    raise AssertionError("raiser should always raise")  # pragma: no cover
```

Comment:
Again the comment about `-> BaseException | None` -- if mypy requires you explicitly return `None` then nvm on that.